test/ParserTest.php

```diff
@@ -119,4 +119,22 @@ public function testCancelReturnsInternalBuffer() {
 
         $this->assertSame("d", $parser->cancel());
     }
+
+    public function testIsValidOnValidParser() {
+        $parser = new Parser((function () {
+            yield 3;
+        })());
+
+        $this->assertTrue($parser->isValid());
+    }
+
+    public function testIsValidOnNonFinishedParser() {
```

Comment:
This is the finished parser, the above method tests with a non-finished parser.

Reply:
As you say, ```testIsValidOnFinishedParser``` is better than ```testIsValidOnNonFinishedParser``` because the test is for checking it's valid on the finished parser.

Reply:
No, it's checking that `isValid()` returns false on a finished parser. One isn't better than the other, they were meant as two names for the two tests.